components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/RemoveUserRolesFunctionImpl.java

```diff
@@ -0,0 +1,131 @@
+     * @return <code>true</code> If the role assigning is successfully completed. <code>false</code> for any other case.
+     */
+    @Override
+    public boolean removeUserRoles(JsAuthenticatedUser user, List<String> removingRoles) {
+
+        if (user != null && removingRoles != null) {
+            try {
+                if (user.getWrapped() != null) {
+                    String tenantDomain = user.getWrapped().getTenantDomain();
+                    String userStoreDomain = user.getWrapped().getUserStoreDomain();
+                    String username = user.getWrapped().getUserName();
+                    UserRealm userRealm = getUserRealm(tenantDomain);
+                    if (userRealm != null) {
+                        UserStoreManager userStore = getUserStoreManager(tenantDomain, userRealm, userStoreDomain);
+                        if (userStore != null) {
+                            userStore.updateRoleListOfUser(
+                                    username,
+                                    removingRoles.toArray(new String[0]),
+                                    new String[0]
+                            );
+                            return true;
+                        } else {
+                            LOG.debug("Null value user store received for the user "
+                                    + user.getMember("username"));
+                        }
+                    } else {
+                        LOG.debug("Null value user realm received for the user "
+                                + user.getMember("username"));
+                    }
+                } else {
+                    LOG.debug("Null value received while getting wrapped content for the user");
+                }
+            } catch (UserStoreException e) {
+                LOG.error("Error in getting user from store at the function ", e);
+            } catch (FrameworkException e) {
+                LOG.error("Error in evaluating the function ", e);
+            }
+        } else {
+            if (user == null) {
+                LOG.error("Invalid value for the user");
+            } else {
+                LOG.error("Invalid value for the new roles");
+            }
+        }
+        return false;
+    }
+
+    private UserRealm getUserRealm(String tenantDomain) throws FrameworkException {
+
+        UserRealm realm;
+        try {
+            realm = AnonymousSessionUtil.getRealmByTenantDomain(UserFunctionsServiceHolder.getInstance()
+                    .getRegistryService(), UserFunctionsServiceHolder.getInstance().getRealmService(), tenantDomain);
+        } catch (CarbonException e) {
+            throw new FrameworkException(
+                    "Error occurred while retrieving the Realm for " + tenantDomain + " to retrieve user roles", e);
+        }
+        return realm;
+    }
+
+    private UserStoreManager getUserStoreManager(String tenantDomain, UserRealm realm, String userDomain)
```

Comment:
This method and method at [1] seems similar. Is it possible to move such methods to a util clas?


[1]https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/37/files#diff-8585d6d2c3d8a5529d73d9087ed7218fR108

Reply:
This might not be possible since module packaging does not allow it at the moment

Reply:
It's fixed